modules/core/src/v2/internal/tssUtils.ts

```diff
@@ -285,4 +340,153 @@ export class TssUtils {
 
     return keychains;
   }
+
+  async signTSSTxRequest(
+    unsignedTx: TransactionRequestJSON,
+    walletPassphrase: string,
+    reqId: RequestTracer
+  ): Promise<any> {
+    const { txRequestId } = unsignedTx;
+
+    const signablePayload = await this.getSignablePayload(unsignedTx);
+
+    const userPShare = await this.getUserPShare(reqId, walletPassphrase);
+
+    const userSignShare = await this.createUserSignShare(signablePayload, userPShare);
+
+    await this.offerUserToBitgoRShare(txRequestId, userSignShare);
+
+    const bitgoToUserRShare = await this.getBitgoToUserRShare(txRequestId);
+
+    const userToBitGoGShare = await this.createUserToBitGoGShare(userSignShare, bitgoToUserRShare, signablePayload);
+
+    await this.sendBitgoToUserGShare(txRequestId, userToBitGoGShare);
+
+    return this.sendTxRequest(txRequestId);
+  }
+
+  async prebuildTxWithIntent(params: PrebuildTransactionWithIntentOptions): Promise<TransactionRequestJSON> {
+    const chain = this.baseCoin.getChain();
+    const intentRecipients = params.recipients.map((recipient) => ({
+      address: { address: recipient.address },
+      amount: { value: `${recipient.amount}`, asset: chain },
+    }));
+
+    const whitelistedParams = {
+      intent: {
+        intentType: params.intentType,
+        sequenceId: params.sequenceId,
+        comment: params.comment,
+        recipients: intentRecipients,
+        memo: params.memo?.value,
+      },
+    };
+
+    const unsignedTx = (await this.bitgo
+      .post(this.bitgo.url('/wallet/' + this.wallet.id() + '/txrequests', 2))
+      .send(whitelistedParams)
+      .result()) as TransactionRequestJSON;
+
+    // TODO(STLX-13411): Implement verify transaction
+    return unsignedTx;
+  }
+
+  async getSignablePayload(unsignedTx: TransactionRequestJSON): Promise<Buffer> {
+    const signablePayload = Buffer.from(unsignedTx.unsignedTxs[0].signableHex, 'hex');
+    const signablePayloadfromPayload = await this.baseCoin.getSignablePayload(unsignedTx.unsignedTxs[0].serializedTx);
+    if (signablePayloadfromPayload.toString('hex') !== signablePayload.toString('hex')) {
+      throw new Error('Missmatched signable payloads');
+    }
```

Comment:
let's leave this part out. i added it while i was debugging tss signing.
ideally, this would be verified in the `verifyTransaction` function _before_ `.send()` is called.